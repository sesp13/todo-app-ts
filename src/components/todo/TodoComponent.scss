.todo-box {
  border: 3px solid black;
  border-radius: 10px;
  padding: 25px;
  margin: 10px;

  .buttons-box {
    display: flex;
    justify-content: center;
    align-items: center;

    button {
      margin-right: 15px;
    }
  }
}
