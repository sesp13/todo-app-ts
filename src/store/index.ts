
export * from './slices'
export * from './store';
export * from './rootReducer';