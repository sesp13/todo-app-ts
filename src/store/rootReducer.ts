import { combineReducers } from "@reduxjs/toolkit";
import { todoSlice } from "./slices";

export const rootReducer = combineReducers({
  todos: todoSlice.reducer
})

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof rootReducer>