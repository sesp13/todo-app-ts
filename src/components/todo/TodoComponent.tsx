import { useDispatch } from 'react-redux';
import { deleteTodo, toggleTodo } from '../../store';
import './TodoComponent.scss';
import { TodoComponentProps } from './TodoComponentProps';

export const TodoComponent = ({ todoItem }: TodoComponentProps) => {
  const dispatch = useDispatch();

  const onToggleTodo = () => {
    const action = toggleTodo(todoItem.id);
    dispatch(action);
  };

  const onDeleteTodo = () => {
    const action = deleteTodo(todoItem.id);
    dispatch(action);
  };

  return (
    <div className="row todo-box">
      <div className="col-6">
        <p>{todoItem.text}</p>
      </div>
      <div className="col-6 buttons-box">
        <button
          aria-label="toggle-btn"
          className="btn btn-primary"
          onClick={() => onToggleTodo()}
        >
          {todoItem.done ? 'Uncomplete' : 'Done'}
        </button>
        <button className="btn btn-secondary">Edit</button>
        <button
          aria-label="delete-btn"
          className="btn btn-danger"
          onClick={() => onDeleteTodo()}
        >
          Delete
        </button>
      </div>
    </div>
  );
};
