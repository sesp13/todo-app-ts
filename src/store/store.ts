import { configureStore } from '@reduxjs/toolkit';
import { saveTodosMiddleware } from './middlewares';
import { rootReducer } from './rootReducer';
import { todoSlice } from './slices/todo';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(saveTodosMiddleware),
  
});

// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;