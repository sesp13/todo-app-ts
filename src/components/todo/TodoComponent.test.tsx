import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../../store';
import { Todo } from '../../types';
import { TodoComponent } from './TodoComponent';

const todoItem: Todo = {
  done: false,
  text: 'My testing todo',
  id: 1,
};

describe('Tests on <TodoComponent />', () => {
  test('should match the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <TodoComponent todoItem={todoItem} />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
