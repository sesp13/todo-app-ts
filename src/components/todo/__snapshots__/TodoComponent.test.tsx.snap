// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <TodoComponent /> should match the snapshot 1`] = `
<div>
  <div
    class="row todo-box"
  >
    <div
      class="col-6"
    >
      <p>
        My testing todo
      </p>
    </div>
    <div
      class="col-6 buttons-box"
    >
      <button
        class="btn btn-primary"
      >
        Done
      </button>
      <button
        class="btn btn-secondary"
      >
        Edit
      </button>
      <button
        class="btn btn-danger"
      >
        Delete
      </button>
    </div>
  </div>
</div>
`;
